src/QuadForm/Lattices.jl

```diff
@@ -966,6 +966,46 @@ function restrict_scalars(L::AbsLat)
   return ZLat(Vabs, Mabs)
 end
 
+@doc Markdown.doc"""
+    restrict_scalars_with_map(L) -> 
+       Tuple{ZLat, Hecke.SpaceRes}}
+
+Given a hermitian lattice `L`, return the $\mathbb Z$-lattice obtained by 
+restriction of scalars, together with the map `f` for extending scalars back.
+"""
+function restrict_scalars_with_map(L)
+  V = ambient_space(L)
+  Vabs, f = restrict_scalars(V, FlintQQ)
+  Babs = absolute_basis(L)
+  Mabs = zero_matrix(FlintQQ, length(Babs), rank(Vabs))
+  for i in 1:length(Babs)
+    v = f\(Babs[i])
+    for j in 1:length(v)
+      Mabs[i, j] = v[j]
+    end
+  end
+  return ZLat(Vabs, Mabs), f
+end
+
+@doc Markdown.doc"""
+    restrict_scalars_with_respect_to_map(L, f) -> ZLat
+
+Given a hermitian lattice `L`, the map `f` for restricting the scalars back,
+return the $\mathbb Z$-lattice obtained by the map `f`.
+"""
+function restrict_scalars_with_respect_to_map(L, f)
+  Vabs = domain(f)
+  Babs = absolute_basis(L)
```

Comment:
Please check that `L` leaves in the codomain of `f`. You should also add a test which fails it `L` does not ( you can use the syntax `@test_throws ArguementError` + a non-working example, in the test file)